rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -1,14 +1,51 @@
+require 'activerecord-import'
+
 module RailsEventStoreActiveRecord
   class EventRepository
     def initialize(adapter: Event)
       @adapter = adapter
     end
     attr_reader :adapter
 
-    def create(event, stream_name)
-      data = event.to_h.merge!(stream: stream_name, event_type: event.class)
-      adapter.create(data)
-      event
+    def append_to_stream(events, stream_name, expected_version)
+      events = [*events]
+      expected_version   = case expected_version
+        when nil
+          raise RubyEventStore::InvalidExpectedVersion
+        when :none
+          -1
+        when :auto
+          eis = EventInStream.where(stream: stream_name).order("position DESC").first
+          (eis && eis.position) || -1
+        else
+          expected_version
+      end
+      in_stream = events.flat_map.with_index do |event, index|
+        position = if expected_version == :any
+          nil
+        else
+          expected_version + index + 1
+        end
+        Event.create!(
```

Comment:
should use the `adapter`

Reply:
I don't see the point of this `adapter` anymore now that the repository needs to work with 2-3 ActiveRecord classes instead of 1.

Reply:
i think it makes it way harder to re-write this repository for another ORM if they are a mixed of multiple active record classes instead of a single responsibility repository which maps to one active record class